modules/core/02-client/keeper/grpc_query.go

```diff
@@ -240,3 +234,40 @@ func (q Keeper) UpgradedClientState(c context.Context, req *types.QueryUpgradedC
 		UpgradedClientState: any,
 	}, nil
 }
+
+// UpgradedConsensusState implements the Query/UpgradedConsensusState gRPC method
```

Comment:
Could you add some comments on this function indicating that the context height should be set for the expected plan height (otherwise this will always return a consensus state not found)

Reply:
Also this will only return a valid consensus state on the last block of the chain and not for any other height

Reply:
> Also this will only return a valid consensus state on the last block of the chain and not for any other height

you can set the height on the grpc req metadata

Reply:
I think this still needs godoc?